File: 0413-arithmetic-slices/0413-arithmetic-slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        if (nums.length < 3) return 0;

        int count = 0;
        int length = 2;

        for (int i = 2; i < nums.length; i++) {
            if (nums[i] - nums[i - 1] == nums[i - 1] - nums[i - 2]) {
                length++;
                count += (length - 2);
            } else {
                length = 2;
            }
        }

        return count;
    }
}

File: 0413-arithmetic-slices/0413-arithmetic-slices.kt
class Solution {
    fun numberOfArithmeticSlices(nums: IntArray): Int {
        if (nums.size < 3) return 0

        var count = 0
        var length = 2

        for (i in 2 until nums.size) {
            if (nums[i] - nums[i - 1] == nums[i - 1] - nums[i - 2]) {
                length++
                count += (length - 2)
            } else {
                length = 2
            }
        }

        return count
    }
}